### PR TITLE
create separate 'development only' requirements file

### PR DESCRIPTION
**Pull request by bhamail**
Collect "development only" requirements into a new file, and use that for various CI tasks.

cc @bhamail / @DarthHater

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+pylint
```
Comment (DarthHater) [inline requirements-dev.txt:1]: Ideally we'd have versions for all these.

**Comment by DarthHater** [comment]
@bhamail I think to generate the requirements.txt it would be good to do the following:

- start a empty venv
- pip install the packages we need
- pip freeze > requirements-dev.txt

This SHOULD get versions to lock to, and also gather any transitive dependencies!

**Comment by DarthHater** [comment]
ALSO, I did some quick reading last night, and people are suggesting instead of `pip` to look at using `pipenv` nowadays, as it has some amount of differentiation for dev dependencies

**Review by DarthHater — CHANGES_REQUESTED** [review]
Suggested change to how we generate the requirements-dev.txt

**Comment by bhamail** [comment]
Added initial `pipenv` changes, but ran into issues getting it to work with CI build. So, I updated the `requirements-dev.txt` file with the versions resolved via `pipenv`. I also added the `pipenv` config files: `Pipfile` and `Pipfile.lock`. I noticed the version of dev dependencies (installed via `pipenv install pylint --dev`) are only locked to versions in the .lock file, but are wildcards in the `Pipfile`.

I think these changes get us closer to using `pipenv`, while also updating to use a shared `dev dependencies` list. We can revisit full conversion to `pipenv` when time permits. (PR's welcome from any `pipenv` experts! :)

**Review by DarthHater — APPROVED** [review]
I'm a lil bit iffy on the Pipenv files going in but I don't see a full issue with doing so.

**Pull request merged**